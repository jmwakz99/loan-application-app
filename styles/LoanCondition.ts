import { Platform, StyleSheet } from "react-native";

import { Styles } from "@/types/global";
import { COLORS } from "@/constants/theme";
import { actuatedNormalize } from "@/utils/responsiveUtils";

const maxAmountTitle: Styles = {
  fontFamily: "Inter-SemiBold",
  fontWeight: "500",
  fontSize: actuatedNormalize(10),
  color: COLORS.dark,
};

const maxLoanAmount: Styles = {
  fontFamily: "Inter-SemiBold",
  fontWeight: "500",
  fontSize: actuatedNormalize(24),
  color: COLORS.primary700,
};

const interestRate: Styles = {
  fontFamily: Platform.OS === "android" ? "Roboto-Bold" : "",
  fontWeight: Platform.OS === "ios" ? "bold" : "700",
  fontSize: actuatedNormalize(11),
  color: COLORS.dark,
};

export const styles = StyleSheet.create({
  maxAmountTitle,
  maxLoanAmount,
  interestRate,
});
