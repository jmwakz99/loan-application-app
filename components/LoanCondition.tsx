import { FC } from "react";
import { Text, View } from "react-native";

import { styles } from "@/styles/LoanCondition";

type Props = {
  maxLoanAmount: number | string;
  interestRate: number;
};

const LoanCondition: FC<Props> = ({ maxLoanAmount, interestRate }) => {
  return (
    <View>
      <Text style={styles.maxAmountTitle}>Maximum Amount:</Text>
      <Text style={styles.maxLoanAmount}>${maxLoanAmount}</Text>
      <Text style={styles.interestRate}>Interest: {interestRate}%</Text>
    </View>
  );
};

export default LoanCondition;
