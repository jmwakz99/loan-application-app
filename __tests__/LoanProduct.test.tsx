import { render } from "@testing-library/react-native";

import LoanCondition from "@/components/LoanCondition";
it("it renders corrrectly", () => {
  const props = {
    maxLoanAmount: 1000,
    interestRate: 5,
    active: false,
  };

  const loanCondition = render(<LoanCondition {...props} />);
  expect(loanCondition).toMatchSnapshot();
});
