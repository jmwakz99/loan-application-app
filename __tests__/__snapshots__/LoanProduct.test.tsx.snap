// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders corrrectly 1`] = `
<View>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 20,
        "fontWeight": "500",
      }
    }
  >
    Maximum Amount:
  </Text>
  <Text
    style={
      {
        "color": "#30C2E3",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 48,
        "fontWeight": "500",
      }
    }
  >
    $
    1000
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "",
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
  >
    Interest: 
    5
    %
  </Text>
</View>
`;
